src/main/java/com/snowflake/kafka/connector/internal/streaming/TopicPartitionChannel.java

```diff
@@ -640,9 +752,12 @@ protected SnowflakeStreamingIngestChannel getChannel() {
     // Every List element corresponds to one record(offset) of a partition
     private final List<Map<String, Object>> tableRows;
 
+    private final List<SinkRecord> sinkRecords;
```

Comment:
This doesn't look good as we're doubling the memory usage, could we reuse the original buffer?

Reply:
Agreed, unfortunately we will have to deal with that since we are buffering. There is no way to send records to DLQ without even buffering them. 

Infact I just realised we format the record into JSON and create new sink records from kafka records. I will fix that in another PR. 

Happy to discuss this more since I am out of ideas on how to regenerate sinkrecords back for buffered records. 

If we dont buffer at all, this issue will solved.

Reply:
Why can't we keep the original SinkRecord and only convert during flushing?

Reply:
Hmm good idea but that might require quite a bit of refactoring.. Can I deal with that in separate PR?

Reply:
sounds good, please add a TODO with JIRA link